Lib/fontTools/pens/hashPointPen.py

```diff
@@ -0,0 +1,57 @@
+# Modified from https://github.com/adobe-type-tools/psautohint/blob/08b346865710ed3c172f1eb581d6ef243b203f99/python/psautohint/ufoFont.py#L800-L838
+import hashlib
+
+from fontTools.pens.pointPen import AbstractPointPen
+from fontTools.pens.transformPen import TransformPointPen
+
+
+class HashPointPen(AbstractPointPen):
+    """
+    This pen can be used to check if a glyph's contents (outlines plus
+    components) has changed.
+
+    Components are decomposed, because that is what happens in TrueType
+    hinting.
+    """
+
+    def __init__(self, glyph, glyphSet=None):
+        self.glyphset = glyphSet
+        self.width = round(getattr(glyph, "width", 0), 9)
+        self.data = ["w%s" % self.width]
+
+    def getHash(self):
```

Comment:
what's the use of `getHash()` method if we also have a `hash` property?

Reply:
A false sense of backwards compatibility I guess; it was called getHash in the Adobe code. But I'll get rid of it, there's no need for it.